feat #1  bridge fixes and tests implementation

lsimone opened the pull request:
bridge modification in order to be consistent with the specification and tests implementation.
The tests aim to verify the behavior of atlas widget used through the bridge and the data binding in both directions (from atlas to hsp and vice-versa).
Close #1

benouat commented on public_html/test/atlasWidgets/multiautocomplete/BaseMultiAutoCompleteTestCase.js:
```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2013 Amadeus s.a.s.
```
Make sure that the License block you use is correct in all files.
Here the year `2013` is wrong, It should be `2014`

benouat commented on public_html/bridge/hsp/atlas.hsp:
```diff
@@ -0,0 +1,96 @@
+// }
+// else{
+            lib.processWidgetMarkup(this.name, out, this.config);
+// }
```
Please make sure the formatting and indentation are correct, even when concerned lines are commented out !

benouat commented on public_html/bridge/hsp/atlas.hsp:
```diff
@@ -0,0 +1,96 @@
+    },
+
+    onConfigChange: function() {
+        this.reloadSection();
```
This method is duplicated. It's already defined on [Line 77](https://github.com/lsimone/atlas-hashspace-bridge/blob/tests-implementation/public_html/bridge/hsp/atlas.hsp#L77)

benouat commented on public_html/bridge/hsp/atlas.hsp:
```diff
@@ -0,0 +1,96 @@
+
+exports.bridge = WidgetBridgeCtrl;
+
+# export template atlas using c:WidgetBridgeCtrl
```
We now support a new template definition syntax that you should use

``` cs
{export template atlas using c:WidgetBridgeCtrl}
...
{/template}
```

benouat commented on public_html/templates/hsp/page.hsp:
```diff
@@ -15,22 +14,30 @@ var PageCtrl = klass({
+        <#atlas wlib="aria.touch.widgets.TouchWidgetLib" name="DoubleSlider" config="{getDoubleSliderConfig(c)}"/>
         {c.data.sliderValue}
     </div>
 # /template
```
Same as previous comment.
We now support a new template definition syntax that you should use

``` cs
{export template page using c:PageCtrl}
...
{/template}
```

benouat commented on public_html/test/atlasWidgets/TemplateTestCaseWrapper.js:
```diff
@@ -0,0 +1,51 @@
+ */
+
+/**
+ * Test suite grouping all tests
```
It's apparently not a _Testsuite_ but more a _Testcase_

benouat commented on public_html/test/atlasWidgets/iconbutton/IconButton.hsp:
```diff
@@ -0,0 +1,32 @@
+    }
+});
+
+# export template page using c:PageCtrl
```
Here again you should change the template definition syntax to

``` cs
{export template page using c:PageCtrl}
...
{/template}
```

benouat commented on public_html/test/atlasWidgets/iconbutton/IconButton.tpl:
```diff
@@ -0,0 +1,24 @@
+         <h2>IconButton</h2>
+         <br />
+
+    {@hsp:Template {
```
Indentation is not respected here !

benouat commented on public_html/test/atlasWidgets/list/List.hsp:
```diff
@@ -0,0 +1,32 @@
+
+# export template page using c:PageCtrl
+    <div class="hsptpl">
+    HASHSPACE OUTER<br/>
```
Please pay attention to the indentation here !

benouat commented on public_html/test/atlasWidgets/list/ListTest.js:
```diff
@@ -0,0 +1,93 @@
+                                delay : 500
+                            });
+                        } else {
+                            this.addItem();
```
This `else` here could be refactored to not be duplicated with the `waitForIE` method

benouat commented on public_html/test/atlasWidgets/list/ListTestTpl.tpl:
```diff
@@ -0,0 +1,43 @@
+         <h2>List</h2>
+         <br />
+
+    {@hsp:Template {
```
Indentation should be respected here

benouat commented on public_html/test/atlasWidgets/multiautocomplete/MultiAuto.hsp:
```diff
@@ -0,0 +1,103 @@
+
+# export template page using c:PageCtrl
+    <div class="hsptpl">
+    HASHSPACE OUTER<br/>
```
Indentation again ! :smile:

benouat commented on public_html/test/atlasWidgets/multiautocomplete/MultiAuto.hsp:
```diff
@@ -0,0 +1,103 @@
+    }
+});
+
+# export template page using c:PageCtrl
```
Update the template declaration syntax

``` cs
{export template page using c:PageCtrl}
...
{/template}
```

benouat commented on public_html/test/atlasWidgets/multiautocomplete/MultiAuto.hsp:
```diff
@@ -0,0 +1,103 @@
+
+var getTextFieldConfig = function(c) {
+    return {
+            label:"Choose your city: ",
```
Try to be consistent in terms of indentation in the same file.
It seems to be `4` spaces. So try to use `4` everywhere

benouat commented on public_html/test/atlasWidgets/multiautocomplete/MultiAutoTpl.tpl:
```diff
@@ -0,0 +1,45 @@
+         <h2>MultiAutoComplete</h2>
+         <br />
+
+    {@hsp:Template {
```
Indentation !!

benouat commented on public_html/test/binding/BindingCSS.tpl.css:
```diff
@@ -0,0 +1,47 @@
+
+
+    {macro main()}
+                .attpl {
```
Indentation !!

benouat commented on public_html/test/binding/BindingInclude.tpl:
```diff
@@ -0,0 +1,38 @@
+{macro main()}
+  <div class="attpl">
+
+  ATLAS INNER<br/>
```
Indentation !!

benouat commented on public_html/test/binding/BindingTemplate.tpl:
```diff
@@ -0,0 +1,54 @@
+    $css: ['test.binding.BindingCSS']
+}}
+    {macro main()}
+TEST ARIA
```
Indentation is not respected here !

benouat commented on public_html/test/binding/page.hsp:
```diff
@@ -0,0 +1,43 @@
+    }
+});
+
+# export template page using c:PageCtrl
```
Please update the template declaration syntax to

``` cs
{export template page using c:PageCtrl}
...
{/template}
```

benouat commented on public_html/test/binding/page.hsp:
```diff
@@ -0,0 +1,43 @@
+
+# export template page using c:PageCtrl
+    <div class="hsptpl">
+    HASHSPACE OUTER<br/>
```
Indentation here !!

benouat commented on public_html/test/hsbridge/MainTestSuite.js:
```diff
@@ -23,5 +23,10 @@ Aria.classDefinition({
+        this.addTests("test.atlasWidgets.iconbutton.IconButtonTest");
+        this.addTests("test.atlasWidgets.list.ListTest");
+        this.addTests("test.atlasWidgets.multiautocomplete.MultiAutoTest");
+//        this.addTests("test.atlasWidgets.tooltip.TooltipTest");
```
Keep indentation in-lined, even with commented line.
Make it consistent

benouat commented on attester.yml:
```diff
@@ -1,20 +1,20 @@
 tests:
  aria-templates:
-  bootstrap: '/aria/bootstrap.js'
+  bootstrap: '/aria/ariatemplates-1.6.0-beta.3.js'
```
Sounds weird to hardcode AriaTemplates version like that.....
Is there any possibility to **not** do this ?

benouat commented:
Review is done for e053d3f

Spotted issues are:
- ariatemplates version hardcoded in `attester` configuration file.
- formatting and indentation lacks of consistency.
- hashspace template declaration to be changed from `# template` to `{template}`.
- couple of small refactor.

---

Globally speaking (that means not at all related to this PR code review)
I clearly doubt about the pertinence of testing the `MultiAutoComplete` in that context, knowing that it would be definitely a better solution to rewrite an hashspace version of that widget...

Commit:
feat #1  bridge fixes and tests implementation

bridge modification in order to be consistent with the specification and tests implementation.
The tests aim to verify the behavior of atlas widget used through the bridge and the data binding in both directions (from atlas to hsp and vice-versa).
Close #1

lsimone commented:
Thank you @benouat 
All the changes have been done, except for ariatemplates version hardcoded in attester configuration file: we should discuss on how to modify it.
